ProjFS.Mac/PrjFSKextTests/MockProc.cpp

```diff
@@ -1,56 +1,130 @@
 #include "MockProc.hpp"
+#include <map>
 #include <string>
 
-static std::string s_procName;
+using std::make_pair;
+using std::map;
+using std::string;
 
-int proc_pid(proc_t)
+static map<uintptr_t /*credential ID*/, int /* UID */> s_credentialMap;
+static map<vfs_context_t /*context*/, int /* UID */> s_contextMap;
+static map<int /*process Id*/, proc> s_processMap;
+static int s_selfPid;
+
+void MockProcess_Reset()
 {
-    return 1;
+    s_processMap.clear();
+    s_credentialMap.clear();
+    s_contextMap.clear();
 }
 
-void proc_name(int pid, char* buf, int size)
+void MockProcess_SetSelfPid(int selfPid)
+{
+    s_selfPid = selfPid;
+}
+
+int proc_pid(proc_t proc)
 {
-    strlcpy(buf, s_procName.c_str(), size);
+    return proc->pid;
 }
 
-void SetProcName(const std::string& procName)
+void proc_name(int pid, char* buf, int size)
 {
-    s_procName = procName;
+    map<int /*process Id*/, proc>::const_iterator found = s_processMap.find(pid);
+    if (found == s_processMap.end())
+    {
+        assert("pid not found in s_processMap");
+    }
+    else
+    {
+        strlcpy(buf, found->second.name.c_str(), size);
+    }
 }
 
 proc_t vfs_context_proc(vfs_context_t ctx)
 {
-    return nullptr;
+    map<vfs_context_t /*context ID*/, int /* UID */>::const_iterator foundPidId = s_contextMap.find(ctx);
+    if (foundPidId == s_contextMap.end())
+    {
+        assert("ctx not found in s_contextMap");
+        return nullptr;
+    }
+
+    return proc_find(foundPidId->second);
```

Comment:
`proc_find` takes a `pid`, but it looks like we're passing in a `UID`.  Should `s_contextMap` be a map from `vfs_context_t` to `PID`?

(This comment is just about naming and comments, it looks like we're only using `s_contextMap` here)

Reply:
Actually, I see that in the tests we refer to the second parameter as "pid":

```
   MockProcess_AddContext(context, 501 /*pid*/);
```

And so it's probably just this file and the header that need to be updated.